Doesn't compile properly on 64-bit Ubuntu 12.10

I have to admit, I'm getting a bit frustrated here: This bloody thing just doesn't work. I tried downloading it and pulling out of the tarball, and it didn't work. I tried configuring, running, making, etc, etc, but no dice. Next, therefore, I downloaded the source files, from here, and tried to run 'make linux'. After installing the relevant dependencies, I made some progress, but now I've hit a wall my technical savvy and google-fu really can't surmount: this just isn't compiling properly. Every time I attempt to make it, I get a stream of 900-odd errors complaining about 'undefined reference to' and then all of the dependencies. Yet more googling reveals that this means that the code is attempting to build _before_ it's connected to the relevant libraries, but I really don't have the know-how to understand the (undocumented) code, and fix it. Please fix this; it's apparently a fairly common issue for programs on various versions on Ubuntu, but it's still very irritating.

--- a/linux.mk
+++ b/linux.mk
@@ -1,6 +1,7 @@
+SHELL := /bin/bash
 CC := gcc
 CFLAGS := -Wall -Werror -std=c99 -pedantic -g -s -Os `pkg-config --cflags gtk+-3.0`
-LFLAGS := `pkg-config --libs gtk+-3.0` -lm -lz
+LFLAGS := `pkg-config --libs gtk+-3.0` -lm -lz -s
 
 OBJ := obj/linux/
 BIN := bin/linux/
@@ -46,14 +47,16 @@ init:
 $(OBJ)%.o: %.c
 	@$(call BEG, "CC", "$<")
 	@mkdir -p $(dir $@)
-	@$(CC) $(CFLAGS) -c $< -o $@ $(ERRORS)
+	@$(CC) -c $< -o $@ $(CFLAGS) $(ERRORS)
 	@$(call END, "CC", "$<")
 
 $(TARGET): $(OBJECTS)
 	@$(call BEG, "LD", "$<")
 	@mkdir -p $(dir $@)
-	@$(CC) $(LFLAGS) $(OBJECTS) -o $(TARGET) -s $(ERRORS)
+	@$(CC) $(OBJECTS) -o $(TARGET) $(LFLAGS) $(ERRORS)
 	@$(call END, "LD", "$<")
 
 clean:
+	@$(call BEGRM, "RM", "$(BIN) $(OBJ)")
 	@$(RM) -rf $(BIN) $(OBJ)
+	@$(call ENDRM, "RM", "$(BIN) $(OBJ)")

--- a/windows.mk
+++ b/windows.mk
@@ -1,6 +1,7 @@
+SHELL := /bin/bash
 CC := i486-mingw32-gcc
 CFLAGS := -Wall -Werror -std=c99 -g -s -Os -mwindows `./pkg-config-script-win --cflags gtk+-2.0` -DGTK2
-LFLAGS := `./pkg-config-script-win --libs gtk+-2.0` -lm -lz -mwindows
+LFLAGS := `./pkg-config-script-win --libs gtk+-2.0` -lm -lz -mwindows -s
 
 WINDRES := i486-mingw32-windres
 
@@ -52,13 +53,13 @@ init:
 $(OBJ)%.o: %.c
 	@$(call BEG, "CC", "$<")
 	@mkdir -p $(dir $@)
-	@$(CC) $(CFLAGS) -c $< -o $@ $(ERRORS)
+	@$(CC) -c $< -o $@ $(CFLAGS) $(ERRORS)
 	@$(call END, "CC", "$<")
 
 $(TARGET): $(OBJECTS) $(OBJ)imagetomap.res
 	@$(call BEG, "LD", "$<")
 	@mkdir -p $(dir $@)
-	@$(CC) $(OBJECTS) $(OBJ)imagetomap.res $(LFLAGS) -o $(TARGET) -s $(ERRORS)
+	@$(CC) $(OBJECTS) $(OBJ)imagetomap.res -o $(TARGET) $(LFLAGS) $(ERRORS)
 	@$(call END, "LD", "$<")
 
 $(OBJ)imagetomap.res: imagetomap.rc
@@ -68,4 +69,6 @@ $(OBJ)imagetomap.res: imagetomap.rc
 	@$(call END, "WINDRES", "$<")
 
 clean:
+	@$(call BEGRM, "RM", "$(BIN) $(OBJ)")
 	@$(RM) -rf $(BIN) $(OBJ)
+	@$(call ENDRM, "RM", "$(BIN) $(OBJ)")

